src/python/pants/engine/internals/build_files.py

```diff
@@ -249,16 +305,55 @@ async def find_target_adaptor(request: TargetAdaptorRequest) -> TargetAdaptor:
             f"TargetAdaptors: {request}"
         )
     address_family = await Get(AddressFamily, AddressFamilyDir(address.spec_path))
-    target_adaptor = address_family.get_target_adaptor(address)
-    if target_adaptor is None:
-        raise ResolveError.did_you_mean(
-            address,
-            description_of_origin=request.description_of_origin,
-            known_names=address_family.target_names,
-            namespace=address_family.namespace,
-        )
+    target_adaptor = _get_target_adaptor(address, address_family, request.description_of_origin)
     return target_adaptor
 
 
+@rule
+async def get_dependencies_rule_action(
+    request: DependenciesRuleActionRequest,
+    maybe_build_file_rules_implementation: MaybeBuildFileDependencyRulesImplementation,
+) -> DependenciesRuleAction:
+    build_file_dependency_rules_class = (
+        maybe_build_file_rules_implementation.build_file_dependency_rules_class
+    )
+    if build_file_dependency_rules_class is None:
+        return DependenciesRuleAction(
```

Comment:
good idea.

Reply:
I've reduced the number of objects created, but since the `address` is part of the response object, I failed to have a singleton. Perhaps make that field optional so it may be `None`? or use a dummy address value for the singleton object?

Reply:
after sleeping on it, went with the `address` may be `None` option.